src/main/kotlin/sh/hsp/techtree/YamlReader.kt

```diff
@@ -1,8 +1,20 @@
 package sh.hsp.techtree
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import java.io.File
 import java.net.URL
 
 // Wiliniak
 interface YamlReader {
```

Comment:
Spodziwałbym się raczej implementacji interfejsu przez konkretną klasę niż `default method`. 

Z tego co wiem `default method` w interfejsach jest używany raczej do dodawania funkcjonalności do **istniejących** interfejsów bez wymuszania zmian przez klasy implementujące te interfejsy, kompatybilność wsteczna i kiedy implementacja takiej metody jest kompletnie obvious (jak np. dodawanie dwóch integerów). 

Zakładamy tutaj, że by default używamy jacksona i że URL jest ścieżką na dysku -> niekoniecznie musi być, jeśli YAML byłby zaciągany po http to ta implementacja się nie sprawdzi, więc spodziwałbym się raczej osobnego pliku w stylu: 

```
class FileSystemYamlReader: YamlReader {
    override fun readToModel(resource: URL): TreeModel {
        val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
        val file = File(resource.toURI())
        val yamlData = file.readText()

        return mapper.readValue(yamlData)
    }
}
```

Reply:
Tutaj chyba nawet możemy mieć inną implementację, żeby yaml był też czytany z `System.in` :smile: